### ISSUE TITLE
There is no way to get the product price including and excluding tax in GraphQL at the same time

### ISSUE THREAD

**Issue by jesse-deboer** [magento/magento2#32437]
### Preconditions (*)

Magento 2.4.1

### Steps to reproduce (*)

1. Go to the admin, then `Stores > Settings > Configuration`
2. Under `Sales > Tax > Price Display Settings` set  `Display Product Prices in Catalog` to `Including and Excluding Tax`
3. Run the following GraphQl query

```
productDetail: products(filter: { sku: {eq: "aRandomProductSku" } }) {
        items {
            price_range {
                minimum_price {
                    regular_price {
                        currency
                        value
                    }
                    final_price {
                        currency
                        value
                    }
                }
                maximum_price {
                    regular_price {
                        currency
                        value
                    }
                    final_price {
                        currency
                        value
                    }
                }
            }
        }
    }
}
```

### Expected result (*)

I would expect there to be a `value_with_tax` and `value_without_tax`, but there is only one value in GraphQL. Since `PriceAdjustment` is deprecated there is no way to get the tax. This means there is no way to display the price with and without tax through GraphQL

### Actual result (*)

There is only a value being returned by GraphQL with tax when you set `Display Product Prices in Catalog` to `Including and Excluding Tax`

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Golf** [comment]
Duplicate #[31313](https://github.com/magento/magento2/issues/31313)

**Comment by oneserv-heuser** [comment]
Any updates on this? We would need this feature.

Also I dont think that this is an duplicated of #31313 as in #31313 it's requested to add the storeConfig for taxes to graphql and here @jesse-deboer wants to get both the price with and without tax in the ProductInterface.

**Comment by cpartica** [comment]
@jesse-deboer Can you elaborate more on why do you need both with vat and not in the catalog?
When we deprecated PriceAdjustment the consideration was that Catalog would not need this because the use cases for catalog and cart are different.
Also, vat is just a type of tax. I wouldn't name the fields "vat" as is very Europe-specific. 
Whatever we do from this point on the schema has to be generalistic for all countries and use cases.
Can you tell us your use case? Are you just trying to have the same data as the cart would have in the catalog?

**Comment by oneserv-heuser** [comment]
@cpartica I can give you a use case. In our frontend we have a "price switcher" where the customer can select whether he wants to see the prices with tax or without tax. When a customer switches prices I dont want to reload the entire page just for that. With the implementation of this feature I could just replace the prices dynamically.

**Comment by jesse-deboer** [comment]
> @jesse-deboer Can you elaborate more on why do you need both with vat and not in the catalog? When we deprecated PriceAdjustment the consideration was that Catalog would not need this because the use cases for catalog and cart are different. Also, vat is just a type of tax. I wouldn't name the fields "vat" as is very Europe-specific. Whatever we do from this point on the schema has to be generalistic for all countries and use cases. Can you tell us your use case? Are you just trying to have the same data as the cart would have in the catalog?

Sorry my mistake, I wil change it to tax indeed.

So normally without GraphQL setting the Display Product Prices in Catalog to Including and Excluding Tax would show both prices, but this doesn't make any change to GraphQL. I think this is the biggest problem without even needing a use case, because people are setting in to both and GraphQL gives you back something else.

Also the use case is in the docs for example on the product page were you can see both prices including and excluding tax at the same time https://docs.magento.com/user-guide/tax/display-settings.html#step-1-configure-catalog-prices-display-settings

Showing both including and excluding in the Netherlands is something that happens very often because we have to show prices with tax (by law) to consumers but sometimes want to show prices also without tax if the visitor is not a consumer but a company.

Overall it just feels weird that changing the setting doesn't do anything with the pricing in GraphQL if you put it on both with and without tax. Also what @oneserv-heuser is mentioning is a great use case.

**Comment by ArjenMiedema** [comment]
@cpartica I currently have several projects (and also had a lot of projects in the past) that require both the prices excluding and including tax in the front end. I'm surprised that this issue is marked as both `Priority P3` and `On Hold` while this feature is something that has been in Magento since the beginning of Magento 1. I'd expect this to be a key feature in GraphQL as well, especially with more and more power is moved to PWA Studio that is all GraphQL as well.

Can you please ensure that this get more priority? I totally agree with the cases mentioned easlier.

**Comment by cpartica** [comment]
I see. 
well we would have to invent a schema for it. Current price schema wasn't meant to support both for Catalog and we had this initiative from product about 3 years ago.
Can you suggest a good schema?
Note:
value_with_tax and value_without_tax would be ok just when you have this setting, but not for the other settings
I wouldn't like to return null either.
You can find me on slack, form a group and discuss about this in order to speed it up

**Comment by srokatonie** [comment]
I totally agree. There's actually no way to query for both prices (incl. and excl. VAT) using GraphQL!

**Comment by MarianNapi** [comment]
Any updates?

**Comment by MarianNapi** [comment]
@engcom-Alfa if you put this ticket "On hold" state you might want to take a look 2 years (!) later?

**Comment by derror** [comment]
We need this feature as well. It's hard to believe that, at the moment, there is no chance to obtain prices from GraphQL excluding VAT.

**Comment by engcom-Delta** [comment]
Hi @jesse-deboer ,

Thanks for your reporting and collaboration.
We have tried to reproduce the issue in Latest 2.4-develop instance and the issue is still reproducible. Kindly refer the screenshots.

**Steps to reproduce**
1. Go to the admin, then Stores > Settings > Configuration
2. Under Sales > Tax > Price Display Settings set Display Product Prices in Catalog to Including and Excluding Tax
3. Run the following GraphQl query
query {
  products(
    filter: { 
      sku: { eq: "Test-Simple" } 
    }
  ) {
    items {
      special_price
      price {
        minimalPrice {
          amount {
            value
          }
        }
        maximalPrice {
          amount {
            value
          }
        }
        regularPrice {
          amount {
            value
          }
        }
      }
      price_range {
        minimum_price {
          regular_price {
            value
            currency
          }
          final_price {
            value
          }
          discount {
            percent_off
            amount_off
          }
        }
        maximum_price {
          regular_price {
            value
            currency
          }
          final_price {
            value
            currency
          }
        }
      }
    }
  }
}
4. Observe , There is only a value being returned by GraphQL with tax when you set Display Product Prices in Catalog to Including and Excluding Tax

![Image](https://github.com/user-attachments/assets/e67f2b36-a55d-4d6e-a7e2-9565a92bb0e0)

![Image](https://github.com/user-attachments/assets/2ed445bb-083a-4298-bb6a-806c1dc3744c)

Thanks.

**Comment by theuargb** [comment]
https://github.com/magento I am working on this

**Comment by theuargb** [comment]
@magento I am working on this

**Comment by theuargb** [comment]
@magento I am not working on this anymore
